src/msg/async/AsyncConnection.h

```diff
@@ -130,10 +130,12 @@ class AsyncConnection : public Connection {
     policy.lossy = true;
   }
 
- entity_addr_t get_peer_socket_addr() const override {
-   return target_addr;
- }
+  entity_addr_t get_peer_socket_addr() const override {
+    return target_addr;
+  }
 
+  void cancel_ops(const set<ceph_tid_t> &ops) override;
```

Comment:
Consider alternatives to std::set here. It pulls in a big dynamically allocated tree for what may not even live that long. boost::container::flat_set has the same semantics and performs well (assuming you insert things in order) and isn't nearly as Heapy.